context_client_test.go

```diff
@@ -0,0 +1,296 @@
+package heimdall
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHTTPClientWithContextSuccess(t *testing.T) {
```

Comment:
You could add a test where the `Do` fails before `ctx.Deadline` to check for ctx deadline exceeded error.